src/AppInstallerCLICore/Workflows/WorkflowBase.h

```diff
@@ -29,6 +29,17 @@ namespace AppInstaller::CLI::Workflow
         PostExecution = 5000,
     };
 
+    enum class SearchPurpose
+    {
+        Default,
+        Upgrade,
+        Install,
+        Uninstall,
+        Export,
+        List,
+        RequiresInstalledSource

```

Comment:
I think RequiresInstalledSource does not fit with SearchPurpose, it should be deduced from other SearchPurpose

Reply:
I wasn't entirely sure how to handle this, actually, because of the `--manifest` flag for upgrade. If the `--manifest` flag is used, then there must be an installed package which meets the upgrade criteria, so EnsureMatchesFromSearchResult (in theory) should show the `No installed package found matching . . .` message rather than `No available upgrades`. In my mind, the search purpose for the `--manifest` flag was specifically to require the installed source, rather than searching for an upgrade, hence the enum value.

If it's okay to show `No available upgrades` when the `--manifest` flag is present and no installed application is found, then I can certainly remove this.